pallets/ethereum-xcm/src/transactional_processor.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019-2025 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.$
+
+use frame_support::storage::{with_transaction, TransactionOutcome};
+use sp_runtime::DispatchError;
+use xcm::latest::prelude::*;
+use xcm_executor::traits::ProcessTransaction;
+
+environmental::environmental!(IS_EVM_REVERT: bool);
```

Comment:
This is a global set for the duration of a block authoring lifecycle, right ?

Reply:
I see, just checked the environmental crate, so it is thread-safe and scoped per the call to `IS_EVM_REVERT::using`.